src/core/errors/errors.ts

```diff
@@ -9,3 +9,17 @@ export class DSNPError extends Error {
     this.name = this.constructor.name;
   }
 }
+
+/**
+ * InvalidHexadecimalSerialization indicates an attempt to coerce a string to hexadecimal serialization failed.
+ */
+export class InvalidHexadecimalSerialization extends DSNPError {
+  outcomeValue: string;
+  originalValue: unknown;
+
+  constructor(originalValue: unknown, outcomeValue: string) {
+    super(`Invalid Hexadecimal Serialization: ${outcomeValue}`);
```

Comment:
What's done with `originalValue`? Looks like it's unused.

Reply:
It's mostly to pass it through for if people want to see what the non-stringified version is